src/library_syscall.js

```diff
@@ -810,7 +857,15 @@ var SyscallsLibrary = {
   },
   __syscall180: function(which, varargs) { // pread64
     var stream = SYSCALLS.getStreamFromFD(), buf = SYSCALLS.get(), count = SYSCALLS.get(), zero = SYSCALLS.getZero(), offset = SYSCALLS.get64();
+#if EMTERPRETIFY_ASYNC
```

Comment:
perhaps one way to reduce code duplication in places like this is if we always passed the extra parameter to `FS.read`, but it was just ignored when not async. And, if we automatically added the code to add the handle, for a list of async-aware syscalls, by modifying the text. We already do similar things for `SYSCALL_DEBUG` at the bottom of this file.